compass/mapper/mapper_test.go

```diff
@@ -0,0 +1,175 @@
+package mapper
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ossf/gemara/layer2"
+	"github.com/ossf/gemara/layer4"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/complytime/complybeacon/compass/api"
+)
+
+// mockMapper is a test implementation of the Mapper interface
+type mockMapper struct {
+	id    ID
+	plans map[string][]layer4.AssessmentPlan
+}
+
+func (m *mockMapper) PluginName() ID {
+	return m.id
+}
+
+func (m *mockMapper) Map(evidence api.RawEvidence, scope Scope) (api.Compliance, api.Status) {
+	return api.Compliance{
+			Catalog:      "test-catalog",
+			Control:      evidence.PolicyId,
+			Category:     "test-category",
+			Requirements: []string{"req-1"},
+			Standards:    []string{"NIST-800-53"},
+		}, api.Status{
+			Title: api.Pass,
+			Id:    &[]api.StatusId{api.N1}[0],
+		}
+}
+
+func (m *mockMapper) AddEvaluationPlan(catalogId string, plans ...layer4.AssessmentPlan) {
+	if m.plans == nil {
+		m.plans = make(map[string][]layer4.AssessmentPlan)
+	}
+	m.plans[catalogId] = plans
+}
+
+func TestNewID(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		expected ID
+	}{
+		{
+			name:     "basic string",
+			input:    "test-mapper",
+			expected: ID("test-mapper"),
+		},
+		{
+			name:     "empty string",
+			input:    "",
+			expected: ID(""),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := NewID(tt.input)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+// TestSet tests map operations (add, retrieve, delete) on the Set type
+func TestSet(t *testing.T) {
+	set := make(Set)
+	assert.NotNil(t, set)
+	assert.Empty(t, set)
+
+	policyMapper := &mockMapper{id: "policy-mapper"}
+	controlMapper := &mockMapper{id: "control-mapper"}
+
+	// Add mappers
+	set[ID("policy-mapper")] = policyMapper
+	set[ID("control-mapper")] = controlMapper
+
+	assert.Len(t, set, 2)
+	assert.Contains(t, set, ID("policy-mapper"))
+	assert.Contains(t, set, ID("control-mapper"))
+	assert.Equal(t, policyMapper, set[ID("policy-mapper")])
+	assert.Equal(t, controlMapper, set[ID("control-mapper")])
+
+	// Retrieve existing mapper
+	retrieved, exists := set[ID("policy-mapper")]
+	assert.True(t, exists)
+	assert.Equal(t, policyMapper, retrieved)
+
+	// Retrieve non-existent mapper
+	_, exists = set[ID("non-existent")]
+	assert.False(t, exists)
+
+	// Delete mapper
+	delete(set, ID("policy-mapper"))
+	assert.Len(t, set, 1)
+	assert.NotContains(t, set, ID("policy-mapper"))
+	assert.Contains(t, set, ID("control-mapper"))
+}
+
+func TestScope(t *testing.T) {
+	scope := make(Scope)
+	assert.NotNil(t, scope)
```

Comment:
Wondering if we really need these assertions since we are just initializing an empty struct?